packages/quantic/force-app/main/default/lwc/quanticFeedbackModal/quanticFeedbackModal.js

```diff
@@ -0,0 +1,234 @@
+import feedbackFormTemplate from './feedbackForm.html';
+import successTemplate from './success.html';
+
+/**
+ * The `QuanticFeedbackModal` component overlays a message modal on top of the current app window, the modal contains a form that allows the user to give feedback.
+ *
+ * @category Search
+ * @category Insight Panel
+ */
+export default class QuanticFeedbackModal extends LightningModal {
+  /**
+   * @api
+   * The list of options to be displayed in the modal.
+   * @type {Array<{label: string, value: string, withDetails: boolean, detailsRequired: boolean}>}
+   */
+  @api options;
+  /**
+   * @api
+   * The function that will be executed when the feedback is submitted.
+   * @type {function}
+   */
+  @api submitFeedback;
+
+  labels = {
+    done,
+    cancel,
+    sendFeedback,
+    explainWhy,
+    selectOneOfOptions,
+    selectTheReason,
+    whatWasTheReason,
+    fillOutField,
+    yourResponsesHelp,
+    thankYouForYourFeedback,
+  };
+
+  /** @type {string} */
+  feedbackValue;
+  /** @type {string} */
+  detailsValue;
+  /** @type {boolean} */
+  feedbackSubmitted = false;
+  /** @type {boolean} */
+  detailsInputError = false;
+  /** @type {string} */
+  error;
+  /** @type {boolean} */
+  validated = false;
+
+  connectedCallback() {
+    this.validateFeedbackModal();
+  }
+
+  /**
+   * Validates the properties provided to the feedback modal.
+   */
+  validateFeedbackModal() {
+    getBueno(this).then(() => {
+      this.validateTheOptionsProperty();
+      if (!this.error) {
+        this.validateOptions();
+      }
+      if (!this.error) {
+        this.validateTheSubmitFeedbackProperty();
+      }
+      this.validated = true;
+    });
+  }
+
+  validateTheOptionsProperty() {
+    if (!Bueno.isArray(this.options)) {
+      console.error('The options provided are not in a valid array.');
+      this.setError();
+    } else if (!this.options.length) {
+      console.error('At least one option must be specified.');
+      this.setError();
+    }
+  }
+
+  validateOptions() {
+    const missingValueOrLabel = this.options?.some(
+      ({value, label}) => !value || !label
+    );
+    if (missingValueOrLabel) {
+      console.error(
+        'Each option requires a label and a value to be specified.'
+      );
+      this.setError();
+    } else {
+      this.options.forEach(({value, label}) => {
+        if (!Bueno.isString(value)) {
+          console.error(`The "${value}" value is not a valid string.`);
+          this.setError();
+        }
+        if (!Bueno.isString(label)) {
+          console.error(`The "${label}" label is not a valid string.`);
+          this.setError();
+        }
+      });
+    }
+  }
+
+  validateTheSubmitFeedbackProperty() {
+    if (typeof this.submitFeedback !== 'function') {
+      console.error('The submitFeedback property is not a valid function.');
+      this.setError();
+    }
+  }
+
+  setError() {
+    this.error = 'c-quantic-feedback-modal Error';
```

Comment:
What is this exactly?

Reply:
it's sets the error message to be displayed when one of the props is invalid.